collectors/PropertiesCollector.py

```diff
@@ -60,6 +65,18 @@ def collect(self):
                 metric_data = value_entry.get('data', [False])[0]
                 metric_value = value_entry.get('values', [False])[0]
 
+                if statkey in self.nested_value_metric_keys:
+                    add_labels, add_label_value_list, add_value = self.unlock_nested_values(statkey, metric_value)
+                    if add_labels[0] not in metrics[statkey]['gauge']._labelnames:
```

Comment:
This is... hacky :>.

Add the `label_names` before L35, when `self.generate_metrics(label_names=self.label_names)` is called. It should not rely on the internal data structure. You can check on `self.nested_value_metric_keys` and extend the list. You've got to sort out, which metric is affected. Maybe it is even needed to do it in the `BaseCollector` and add a flag to pass through.

Reply:
True, but we also use the internal variable further down in the collectors. We can wrap this in a function so that it doesn't appear so prominently in the code.

Reply:
Latest commit reflects this idea.

Reply:
Yeah I like this more. The use of `._labelnames` happened during the rewrite a very while back. Wasn't aware of this anymore.